### ISSUE TITLE
Enable ssh_multiplexing on same connection settings in parallel mode cause hang with CPU 100%

### ISSUE THREAD

**Issue by ihipop** [deployphp/deployer#1509]
| Q                 | A
| ----------------- | ---
| Issue Type        | Bug
| Deployer Version  | 6.0.5
| Local Machine OS  | `Debian SID`  or  `alpine 3.4` in docker image from  [`ihipop/deployer-toolkit:dev`](https://hub.docker.com/r/ihipop/deployer-toolkit/tags/)
| Remote Machine OS | N/A

### Description 
Enable ssh_multiplexing on same connection settings in parallel mode cause hang with CPU 100%

### Steps to reproduce

1.  Set different host in host config ,they **have different deploy path**,but **on the same server**,and **use same connection settings,** then open  parallel mode,after deployer connected one host, it hangs with CPU 100% when connect  to another

1. Set `set('ssh_multiplexing', false);` in `deploy.php`, It works fine  in parallel mode .

1. Don't do `Step 2`, Just  Change the `hostname` connection settings in `hosts.yml`, ,eg: change `bugtest-2`.`hostname` to another domain `TEST2.DOMAIN`,**but referrer to the same host ip address** , It woks fine in parallel mode .

### Content of `deploy.php`
```php
set('ssh_multiplexing', true);
```
### Content of `hosts.yml`
```yml
.base:  &base
    user: root
    port: 2222
    sshOptions:
        UserKnownHostsFile: /dev/null
        StrictHostKeyChecking: no

bugtest-1:
    hostname: TEST.DOMAIN
    <<: *base
    stage: test
    roles:
        - app
    deploy_path:  /tmp/bugtest-1/

bugtest-2:
    hostname: TEST.DOMAIN
    <<: *base
    stage: test
    roles:
        - app
    deploy_path: /tmp/bugtest-2/
#    branch: develop
```

### Output log
![image](https://user-images.githubusercontent.com/423077/35146489-262f8174-fd46-11e7-9030-8587d44c8948.png)

![image](https://user-images.githubusercontent.com/423077/35146517-39311288-fd46-11e7-87fc-0e94ec88e569.png)

### Conclusion
I think it's because the conflict of `ControlPath` , Suggest to have `host name`,eg: `bugtest-2` when building `ControlPath` in parallel mode

https://github.com/deployphp/deployer/blob/15274524b28c94e5967fa509a91680f599ddd23d/src/Ssh/Arguments.php#L128-L157

**Comment by antonmedv** [comment]
Agree. Looks like a bug.

**Comment by antonmedv** [comment]
Can'tWill be fixed with 7.0 after refactoring hostname config params.

**Comment by antonmedv** [comment]
Fixed in #1904
